blocks/library/categories/data.js

```diff
@@ -0,0 +1,13 @@
+/**
+ * Returns a Promise with the categories or an error on failure.
+ *
+ * @returns {wp.api.collections.Categories} Returns a Promise with all categories.
+ */
+export function getCategories() {
+	const categoriesCollection = new wp.api.collections.Categories();
+
+	const categories = categoriesCollection.fetch();
```

Comment:
I see. I was confusing this `getCategories` function with the `getCategories` method on the component, and the latter is called recursively.